nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/csv/CSVRecordReader.java

```diff
@@ -155,4 +159,59 @@ private List<RecordField> getRecordFields() {
     public void close() throws IOException {
         csvParser.close();
     }
+
+    /**
+     * This method builds a text representation of the CSV record by searching character-by-character until the
+     * record separator is found. Because we never want to consume input we don't use, the method attempts to match
+     * the separator separately, and as it is not matched, the characters are added to the returned string.
+     * @param reader the Reader providing the input
+     * @param recordSeparator the String specifying the end of a record in the input
+     * @return a String created from the input until the record separator is reached.
+     * @throws IOException if an error occurs during reading
+     */
+    protected String readNextRecord(Reader reader, String recordSeparator) throws IOException {
+        int indexIntoSeparator = 0;
+        int recordSeparatorLength = recordSeparator.length();
+        StringBuilder lineBuilder = new StringBuilder();
+        StringBuilder separatorBuilder = new StringBuilder();
```

Comment:
Is it necessary to build the separator as opposed to appending the `recordSeparator` to the `lineBuilder`?